migrate.py

```diff
@@ -35,16 +36,39 @@ async def migrate_organization(session):
 
 async def main():
     async with aiohttp.ClientSession() as session:
-        auth_config = await get_existing_auth_config(session)
-        await migrate_organization(session)
-        await migrate_users(session)
-
-        if not auth_config:
-            print("==== Warning ====")
-            print("An existing auth configuration was not found")
-            print("or does not match an existing IdP configuration.\n")
-            print("Please review your auth configuration in the provided")
-            print("page at: {your domain}/cas/admins")
-    print("Migration Complete")
+        mode = await get_auth_mode(session)
+
+    # test connection to CAS
+    if not mode:
+        print("==== Error ====")
+        print("Unable to connect to the Central Auth Service (CAS)\n")
+        print("Please check your Fiftyone Teams deployment and ensure that")
+        print("there is a running CAS at the supplied CAS_BASE_URL\n")
+
+    # we don't want to run in legacy mode
+    if mode == "legacy":
+        print("==== Alert ====")
+        print("The migration script must be run using a Central Auth")
+        print("Service (CAS) configured to internal mode.\n")
+        print("The currently running CAS is configured to legacy mode\n")
+        print("Please check the FIFTYONE_AUTH_MODE environment variable")
+        print("in your running Fiftyone Teams Deployment\n")
+        print("For help, contact your Voxel51 Customer Service Representative\n")
+
+    # it's internal, green light go!
+    if mode == "internal":
+        async with aiohttp.ClientSession() as session:
+            auth_config = await get_existing_auth_config(session)
+            await migrate_organization(session)
+            await migrate_users(session)
+            if not auth_config:
```

Comment:
confirming that you still want to proceed with the migration if the auth_config is undefined? otherwise, we can return early with the warning message

Reply:
yeah I think this is okay; theoretically they could migrate the users and orgs and then update the auth config in the super-admin UI. It doesn't need to happen in a specific order in this case

Reply:
If that becomes a pain point we can always update